server/api/users.js

```diff
@@ -26,30 +31,23 @@ router.put(
     const newName = req.body.name;
     const newPassword = req.body.password;
     const newEmail = req.body.email;
-
-    const currentName = req.user.name;
-    const currentEmail = req.user.email;
-    const currentPassword = req.user.password;
-
-    if (req.user == null || req.user == undefined) {
+    const nameResult = nameValidate(newName);
+    const emailResult = emailValidate(newEmail);
+    const passwordResult = passwordValidate(newPassword);
+    if (!req.use) {
       res.status(401).send("Unauthorised");
       return;
     }
 
-    if (newName === "" || newName.length == undefined || newName == null) {
-      return currentName;
+    if (nameResult == false) {
+      res.status(400).send("Error");
     }
 
-    if (
-      !newEmail.includes("@") ||
-      newEmail === "" ||
-      newEmail === undefined ||
-      newEmail == null
-    ) {
-      return currentEmail;
+    if (emailResult==false) {
+      res.status(400).send("Error");
     }
-    if (newPassword === "" || newPassword == undefined || newPassword == null) {
-      return currentPassword;
+    if (passwordResult==false) {
+      res.status(400).send("Error");
```

Comment:
Try to remove the duplication here, you can group the 3 conditions: `if (!nameResult || !emailResult etc...`

Reply:
it is done, thank you

Reply:
I pushed the new changes